### PR TITLE
Feature/fluent combobox

### PR DESCRIPTION
**Pull request by danwalmsley**
## What does the pull request do?

image

## Checklist

- [ ] Added unit tests (if possible)?
- [ ] Added XML documentation to any related classes?
- [ ] Consider submitting a PR to https://github.com/AvaloniaUI/Avaloniaui.net with user documentation

### CODE REVIEW

File: src/Avalonia.Themes.Fluent/ComboBox.xaml [hunk 1]
```diff
@@ -1,64 +1,191 @@
...
+                    Background="{DynamicResource ComboBoxDropDownBackground}"
+                                BorderBrush="{DynamicResource ComboBoxDropDownBorderBrush}"
+                                BorderThickness="{DynamicResource ComboBoxDropdownBorderThickness}"
+                                Margin="0,-1,0,-1"
+                                Padding="{DynamicResource ComboBoxDropdownBorderPadding}"
+                                HorizontalAlignment="Stretch"
+                                CornerRadius="{DynamicResource OverlayCornerRadius}">
+              <ScrollViewer>
+                <ItemsPresenter Name="PART_ItemsPresenter"
+                                Items="{TemplateBinding Items}"
```
Comment (maxkatz6) [inline src/Avalonia.Themes.Fluent/ComboBox.xaml:125]: I think there is missed Margin. It exists in [WinUI's style](https://github.com/microsoft/microsoft-ui-xaml/blob/master/dev/ComboBox/ComboBox_themeresources.xaml#L695) and has value "0,4,0,4".
```suggestion
                                Items="{TemplateBinding Items}"
                                Margin="{DynamicResource ComboBoxDropdownContentMargin}"
```

**Review by maxkatz6 — COMMENTED** [review]
Something wrong with vertical text alignment in ComboBoxItem.

image

File: src/Avalonia.Themes.Fluent/ComboBoxItem.xaml [hunk 2]
```diff
@@ -1,9 +1,25 @@
...
+  </Design.PreviewWith>
+
   <Style Selector="ComboBoxItem">
-    <Setter Property="Background" Value="Transparent"/>
-    <Setter Property="Padding" Value="2"/>
-    <Setter Property="HorizontalAlignment" Value="Stretch"/>
-    <Setter Property="HorizontalContentAlignment" Value="Left"/>
+    <Setter Property="TextBlock.Foreground" Value="{DynamicResource ComboBoxItemForeground}" />
+    <Setter Property="Background" Value="{DynamicResource ComboBoxItemBackground}" />
+    <Setter Property="Padding" Value="{DynamicResource ComboBoxItemThemePadding}" />
```
Comment (maxkatz6) [inline src/Avalonia.Themes.Fluent/ComboBoxItem.xaml:21]: Interesting moment. I noticed some margin or border between ComboBoxItems in WinUI, but couldn't found it in the [style](https://github.com/microsoft/microsoft-ui-xaml/blob/master/dev/ComboBox/ComboBox_themeresources.xaml#L705). With devtools I found that there is actually border, which is sets in generic.xaml and is not overridden by WinUI. Resources (they are missed in Avalonia in master):
```xaml
<StaticResource x:Key="ComboBoxItemRevealBorderBrush" ResourceKey="SystemControlTransparentRevealBorderBrush" />
<Thickness x:Key="ComboBoxItemRevealBorderThemeThickness">1</Thickness>
```
```suggestion
    <Setter Property="Padding" Value="{DynamicResource ComboBoxItemThemePadding}" />
    <Setter Property="BorderBrush" Value="{ThemeResource ComboBoxItemRevealBorderBrush}" />
    <Setter Property="BorderThickness" Value="{ThemeResource ComboBoxItemRevealBorderThemeThickness}" />
```

**Comment by maxkatz6** [comment]
Except issue with alignment and diffs with border and margin, there is one difference in popup behavior: UWP/WinUI popup opens on top of ComboBox with specific offset, while Popup in Avalonia always shows on bottom. 
From other hand, ComboBox in Avalonia feels better because it doesn't change size of ComboBox, when Popup is open moving everything around.

combobox

File: src/Avalonia.Themes.Fluent/ComboBoxItem.xaml (hunk 2 above)
Comment (maxkatz6) [reply src/Avalonia.Themes.Fluent/ComboBoxItem.xaml:21]: For another controls we also should look at both styles - from generic.xaml and from WinUI, to combine setters from them.

**Review by maxkatz6 — CHANGES_REQUESTED** [review]
@danwalmsley 
Resources is missed
```xaml
<StaticResource x:Key="ComboBoxItemRevealBorderBrush" ResourceKey="SystemControlTransparentRevealBorderBrush" />
<Thickness x:Key="ComboBoxItemRevealBorderThemeThickness">1</Thickness>
```

**Comment by danwalmsley** [comment]
@maxkatz6 seems to be part of reveal resources? https://github.com/microsoft/microsoft-ui-xaml/blob/264f679ae49af3448a84fce75d03e3e838c0a0d5/dev/Materials/Reveal/RevealBrush_rs1_themeresources.xaml 

we can add those when we get reveal in future.

**Comment by maxkatz6** [comment]
@danwalmsley 
Yes, part of reveal (without actual reveal in that case). But they are used with commit
https://github.com/AvaloniaUI/Avalonia/pull/4019/commits/2d9b0c6b439c233e4d46198347d9f0f45b8b630e
So it should be runtime error, isn't it?

**Pull request merged**